### PR TITLE
Add templates and logic to allow PIF/USDC/etc. postings outside of JoinTTS's site

### PR DESCRIPTION
**Pull request by geekygirlsarah**
[:sunglasses: PREVIEW](https://federalist-proxy.app.cloud.gov/preview/18f/join.tts.gsa.gov/add-pif-template/)

Changes proposed in this pull request:
- Add a TEMPLATE_PIF_USDC.md which describes a PIF/USDC/etc. position
- Template also adds a "application link" which is the URL to the site with the description
- Logic is added to the home page to mix these jobs with the others

This does not add any additional jobs.

/cc @LeighFinkel

### CODE REVIEW

File: pages/index.md [hunk 1]
```diff
@@ -47,8 +47,13 @@ permalink: /
...
           {% unless job.path contains 'performance-profiles' %}
           {% assign info_sessions = job | future_info_sessions_for_job %}
+          {% if job["information link"] %}
+            {% assign link_url = job["information link"] %}
+          {% else %}
+            {% assign link_url = site.baseurl | append: job.url %}
+          {% endif %}
           <li class="{% if info_sessions.size > 0 %}with-info-sessions{% endif %}">
-            {{ infosessions.size }}<a href="{{ site.baseurl }}{{ job.url }}">{{ job.title }}</a>
+            {{ infosessions.size }}<a href="{{ link_url }}">{{ job.title }}</a>
```
Comment (mgwalker) [inline pages/index.md:56]:
```suggestion
            <a href="{{ link_url }}">{{ job.title }}</a>
```

Looks like some of my old debugging cruft stuck around? 😬

**Review by mgwalker — COMMENTED** [review]
No concerns with the code, but I do have two questions:

1. What alternatives did y'all consider besides making a new template file? E.g., it might be possible to modify the `job` template to check the `information link` property and if it's set, render a meta refresh redirect instead of a content page. Then the existing job template would continue to be valid for all cases, and we'd have a simple mechanism for handling any job that's hosted outside of JoinTTS. And what you've done with the index page already handles that too, so most users would never see the redirect.

   - The reason I mention this is I'm worried about there being a bunch of template files at some point and the risk of those diverging so Team Talent ends up being back in the position of multiple confusing templates. But having one all-purpose template is also risky if it makes the template too hard to use. 

2. Assuming no changes related to my first question, do job pages get generated for the new template even though there is no content? I think that's mostly fine, but it will create archive pages with no content. If the pages are being created, there's probably something that could be done in the archiving script to detect that the `information link` property exists and skipping those.

**Review by mgwalker — APPROVED** [review]
I meant to approve this last night. 😬 My questions weren't meant to be blockers or anything like that!

**Comment by geekygirlsarah** [comment]
It's also why I tried to clarify which template was for which, take out the extra stuff in the misc/external template. It also reduces duplication needed for descriptions on the other sites (which was what she brought up as they didn't want to sync changes on both sites).

I had a few meetings, but am going to see if it makes a page too. (I think it does, and not sure how that affects any archiving)

**Comment by geekygirlsarah** [comment]
It looks like it doesn't generate a page, and the archiver doesn't try to archive it, so I might merge this in. (There is a jobs.json that needs updating or removed, and Greg suggests removing, so I'll do that in another issue.)

**Pull request merged**